src/search/structures.go

```diff
@@ -0,0 +1,35 @@
+package search
+
+const DefaultUserAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36"
+
+type Dom struct {
+	Result      string
+	Link        string
+	Title       string
+	Description string
+	NextPage    string
+}
+
+type Options struct {
+	UserAgent      string
+	Limit          int
+	ProxyAddr      string
+	FollowNextPage bool
+}
+
+type Result struct {
+	Rank        int    `json:"rank"`
+	URL         string `json:"url"`
+	Title       string `json:"title"`
+	Description string `json:"description"`
+}
+
+func (r Result) Hash() string {
```

Comment:
Will this be used with a HashSet, or are we making a new (better) solution for removing duplicate searches?

Reply:
Lets just use the default go `map`. It's more clear what it does, we don't add the dependency, and there is going to be *marginally* more code.